pint/models/polycos.py

```diff
@@ -0,0 +1,612 @@
+            errorMssg = 'Format name \''+formatName+ '\' is already exist. '
+            raise Exception(errorMssg)
+
+        pFormat = {'format' : formatName}
+
+        if methodMood == 'r':
+            if readMethod == None:
+                raise BaseException('Argument readMethod should not be \'None\'.')
+            
+            pFormat['read_method'] = readMethod
+            pFormat['write_method'] = writeMethod
+            registry.register_reader(pFormat['format'], table.Table, 
+                                    pFormat['read_method'])
+        elif methodMood == 'w':
+            if writeMethod == None:
+                raise BaseException('Argument writeMethod should not be \'None\'.')
+           
+            pFormat['read_method'] = readMethod
+            pFormat['write_method'] = writeMethod
+            registry.register_writer(pFormat['format'], table.Table, 
+                                    pFormat['write_method'])
+        elif methodMood == 'rw':
+            if readMethod == None or writeMethod == None:
+                raise BaseException('Argument readMethod and writeMethod' 
+                                    'should not be \'None\'.')
+
+            pFormat['read_method'] = readMethod
+            pFormat['write_method'] = writeMethod
+
+            registry.register_reader(pFormat['format'], table.Table, 
+                                    pFormat['read_method'])
+            registry.register_writer(pFormat['format'], table.Table, 
+                                    pFormat['write_method'])
+
+        self.polycoFormat.append(pFormat)
+        
+
+    def generate_polycos(self, mjdStart, mjdEnd, parFile, obs, 
+                         segLength, ncoeff, obsFreq, maxha,
+                         method = "TEMPO",numNodes = 20):
+        """
+        Generate the polyco file data file. 
+
+        Parameters
+        ---------
+        mjd : 
+
+        """
+        self.read_parfile(parFile)   # Read Parfile
+        mjdStart = np.longdouble(mjdStart)*u.day
+        mjdEnd = np.longdouble(mjdEnd)*u.day
+        timeLength = mjdEnd-mjdStart
+        segLength = np.longdouble(segLength)*u.min
+        obsFreq = float(obsFreq)
+        month = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug',
+                 'Sep','Oct','Nov','Dec']
+        # Alocate memery
+        coeffs = np.longdouble(np.zeros(ncoeff))
+        entryList = []
+        entryIntvl = np.arange(mjdStart.value,mjdEnd.value,
+                                segLength.to('day').value)
+        if entryIntvl[-1] < mjdEnd.value:
+            entryIntvl = np.append(entryIntvl, mjdEnd.value) 
+
+        # Make sure the number of nodes is bigger then number of coeffs.
+        if numNodes < ncoeff:
+            numNodes = ncoeff+1   
+    	# generate the ploynomial coefficents
+    	if method == "TEMPO":
+            
+           
+    	# Using tempo1 method to create polycos
+            for i in range(len(entryIntvl)-1):
```

Comment:
This For loop is under the if  method == "TEMPO" statement. I believe is should be right